identity/src/main/java/com/android/identity/mdoc/engagement/EngagementParser.java

```diff
@@ -172,9 +172,13 @@ void parse(@NonNull byte[] encodedEngagement) {
                 if (Util.cborMapHasKey(map, 5)) {
                     List<DataItem> originInfoItems = Util.cborMapExtractArray(map, 5);
                     for (DataItem oiDataItem : originInfoItems) {
-                        OriginInfo originInfo = OriginInfo.decode(oiDataItem);
-                        if (originInfo != null) {
-                            mOriginInfos.add(originInfo);
+                        try {
+                            OriginInfo originInfo = OriginInfo.decode(oiDataItem);
+                            if (originInfo != null) {
+                                mOriginInfos.add(originInfo);
+                            }
+                        } catch (Exception e) {
+                            Logger.w(TAG, "OriginInfo is incorrectly formatted. We expect OriginInfo = {\"cat\": uint, \"type\": uint, \"details\": tstr} in accordance with WD6 of 18013-7.");
```

Comment:
This is overly verbose. Just "OriginInfo is incorrectly formatted" also please include `e` in the `Logger.w()` call.